src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -114,7 +115,8 @@
    :version s/Str
    :environment (s/maybe s/Str)
    :transaction_uuid (s/maybe s/Str)
-   :producer_timestamp (s/maybe pls/Timestamp)
+   :producer_timestamp (s/pred #(not (nil? (to-timestamp %))) 'coercable-to-timestamp)
```

Comment:
Well, it gives a more useful error message on bad input, which is the entire point of having a schema around.